torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,19 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        optim_state_dict = self.optim.state_dict()
+        if optim_state_dict is not None and self.averager.step is not None:
+            optim_state_dict['step'] = self.averager.step
+        return optim_state_dict
 
     def load_state_dict(self, state_dict):
         self.optim.load_state_dict(state_dict)
+        if 'step' in state_dict.keys():
+            self.averager.step = state_dict['step']
+        else:
+            warn("Loaded state dict does not contain a step counter for an averager. " +
+                 "Setting step counter to 0.")
+            self.averager.step = 0
```

Comment:
cc: @rohan-varma Is this the desired behavior? This seems reasonable to me.

If so, we should add a unit test that checks for a warning here (using `self.assertWarnsRegex(expected_warning=..., expected_regex=...)`. (The test loads a state dict without `"step"` as a top-level key and checks that the warning was issued.)

Reply:
yeah, we discussed this offline. 

might also be good to get @wayi1 who requested this feature feedback: If the saved state_dict did not contain a step, we're defaulting it to 0

Reply:
let's also add a unittest as @awgu suggested above (can be combined with my below unittest suggestion too). 

You can find documentation for assertWarnsRegex here: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarnsRegex

Reply:
Added unittest for warning check